azkaban-hadoop-security-plugin/src/main/java/azkaban/security/HadoopSecurityManager_H_2_0.java

```diff
@@ -462,15 +465,27 @@ public void prefetchToken(final File tokenFile, final Props props, final Logger
     doPrefetch(tokenFile, props, logger, userToProxy);
   }
 
+  /*
+   * Create a suffix for Kerberos principal, the format is,
+   * az_<host name>_<execution id><DOMAIN_NAME>
+   */
+  private String kerberosSuffix(final Props props) {
+    final String host = ServiceProvider.SERVICE_PROVIDER.getInstance(Props.class)
+        .getString(AZKABAN_SERVER_HOST_NAME, "unknown");
```

Comment:
Isn't the `AZKABAN_SERVER_HOST_NAME` property available in the Props instance passed in as a parameter to the kerberosSuffix method? Why do we need to get another instance?

Reply:
The name in AZKABAN_SERVER_HOST_NAME is not fully qualified. What we send in KAfka event is fully qualified.
Inorder to perform a join, we need to match the names.

Reply:
talked offline it seems AZKABAN_SERVER_HOST_NAME is not in the props instance passed in as a parameter. 
Could you add a comment explaining it?